nav2_costmap_2d/include/nav2_costmap_2d/observation_buffer.hpp

```diff
@@ -49,6 +49,12 @@
 #include "nav2_costmap_2d/observation.hpp"
 #include "nav2_util/lifecycle_node.hpp"
 
+/**
+ * private field 'tf_tolerance_' is not yet used which causes compilation failure
+ * when using flag -Werror with Clang 12
+ */
+#pragma GCC diagnostic ignored "-Wunused-private-field"
```

Comment:
Rather than ignoring an unused variable, can we use it in the tf2 buffer calls or delete it?